account_payment_term_extension/models/account_payment_term.py

```diff
@@ -0,0 +1,253 @@
+                continue
+            try:
+                payment_days = record._decode_payment_days(record.payment_days)
+                error = any(day <= 0 or day > 31 for day in payment_days)
+            except Exception:
+                error = True
+            if error:
+                raise exceptions.Warning(_("Payment days field format is not valid."))
+
+    payment_days = fields.Char(
+        string="Payment day(s)",
+        help="Put here the day or days when the partner makes the payment. "
+        "Separate each possible payment day with dashes (-), commas (,) "
+        "or spaces ( ).",
+    )
+
+
+class AccountPaymentTerm(models.Model):
+    _inherit = "account.payment.term"
+
+    sequential_lines = fields.Boolean(
+        string="Sequential lines",
+        default=False,
+        help="Allows to apply a chronological order on lines.",
+    )
+    holiday_ids = fields.One2many(
+        string="Holidays",
+        comodel_name="account.payment.term.holiday",
+        inverse_name="payment_id",
+    )
+
+    def apply_holidays(self, date):
+        holiday = self.holiday_ids.search(
+            [("payment_id", "=", self.id), ("holiday", "=", date)]
+        )
+        if holiday:
+            return fields.Date.from_string(holiday.date_postponed)
+        return date
+
+    def apply_payment_days(self, line, date):
+        """Calculate the new date with days of payments"""
+        if line.payment_days:
+            payment_days = line._decode_payment_days(line.payment_days)
+            if payment_days:
+                new_date = None
+                payment_days.sort()
+                days_in_month = calendar.monthrange(date.year, date.month)[1]
+                for day in payment_days:
+                    if date.day <= day:
+                        if day > days_in_month:
+                            day = days_in_month
+                        new_date = date + relativedelta(day=day)
+                        break
+                if not new_date:
+                    day = payment_days[0]
+                    if day > days_in_month:
+                        day = days_in_month
+                    new_date = date + relativedelta(day=day, months=1)
+                return new_date
+        return date
+
+    def compute(self, value, date_ref=False, currency=None):
+        """Complete overwrite of compute method for adding extra options."""
+        # FIXME: Find an inheritable way of doing this
+        self.ensure_one()
+        last_account_move = self.env.context.get("last_account_move", False)
+        date_ref = date_ref or fields.Date.today()
+        amount = value
+        result = []
+        if not currency:
+            if self.env.context.get("currency_id"):
+                currency = self.env["res.currency"].browse(
+                    self.env.context["currency_id"]
+                )
+            else:
+                currency = self.env.user.company_id.currency_id
```

Comment:
Use self.env.company instead of self.env.user.company_id

Reply:
Changed, thanks.

Reply:
I think you used `self.env.user.company` instead which is causing an error in the tests: https://travis-ci.com/github/OCA/account-payment/jobs/472291835#L1181

Reply:
Indeed, I made the change blindly and very late, hehe. Now it should be OK.

Reply:
Hehe ok :+1: